src/lib.rs

```diff
@@ -1394,7 +1388,7 @@ decl_module! {
             });
 
             // Generate event
-            Self::deposit_event(RawEvent::PostTextUpdated(post.id, <PostById<T>>::get(post_id).text_change_history.len() as u64));
+            Self::deposit_event(RawEvent::PostTextUpdated(post_id, <PostById<T>>::get(post_id).text_change_history.len() as u64));
```

Comment:
Simplify, separate out writing to storage from posting event.

Reply:
accepted.